Have the TelemetryClient be less noisy after being shut down

tbradellis opened the pull request:
This is a start for issue #176 

I think this small change is all that is needed. 
If the TelemetryClient executor is shutdown 
log that the batch was not sent.
log some info about the batch
return before attempting to send.

I will also add a test to this pr.

jkwatson commented on telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java:
```diff
@@ -169,6 +169,10 @@ private void scheduleBatchSend(
   private void sendWithErrorHandling(
       BatchSender batchSender, TelemetryBatch<? extends Telemetry> batch, Backoff backoff) {
+    if (executor.isShutdown()) {
+      LOG.info("TelemetryClient background Executor is shutdown. Batch not sent", batch.toString());
```
I think if we just throw the toString into the log message, it'll end up being potentially huge log line. How about just logging the number of items in the batch, and the type of batch it is (`batch.getClass().getSimpleName()`)?

jkwatson commented:
Looks good as a start (just one litttle comment), and needs a rebase.

tbradellis commented:
After looking at it more, I couldn't think of a way to pull off a test for it.
Switched to use isTerminated and added a try/catch when the batch is scheduled.   
Changed log entry to get just simple class name.

breedx-nr commented on telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java:
```diff
@@ -164,11 +165,21 @@ private void scheduleBatchSend(
 
   private void sendWithErrorHandling(
       BatchSender batchSender, TelemetryBatch<? extends Telemetry> batch, Backoff backoff) {
+    if (executor.isTerminated()) {
```
I don't think we want to check the executor here.  If we have reached this line, it means that the executor has invoked us, and we should be ok to send the batch.  If the timing is perfectly wrong, I think this could cause us to lose data.

breedx-nr commented:
Thanks!  I think this is a good change.
